### DISCUSSION TITLE
I get an error when I try to use the function "convert_to_pdf".

### DISCUSSION THREAD

**Discussion by hoaile-novobi** [pymupdf/PyMuPDF#1967] [Q&A]
Hi admin, please help to find out what happen with my PDF file, my file is a form format PDF and i would like to convert it to flatten format so i use the "convert_to_pdf" function, but it throw an error **RuntimeError('substitute font creation is not implemented yet')**

Here is my code:
    

> attachment = base64.b64encode(image_file.read())
>     pdf_file = fitz.open("pdf", base64.b64decode(attachment))
>     pdf_bytes = pdf_file.convert_to_pdf()

My PDF: https://drive.google.com/file/d/18D-Kn2cRWE_dLRItNgOb4N_ys5CUA-X6/view?usp=sharing
Please help.

**Comment by JorjMcKie** [comment]
This is not a bug, but simply not supported. There are limitations WRT font support.
Similarly, Type3 fonts are also not supported.

**Comment by hoaile-novobi** [comment]
Do you have any idea for the issue? How can i resolve this case?

**Reply by JorjMcKie** [reply]
I just tried to replace the used font with some other, but no success.

This is a PDF already, so what are you trying to achieve by converting it one more time?

**Reply by David-Essaadi** [reply]
I have this same issue. The reason I'm converting PDF -> PDF is because of this issue: https://github.com/pymupdf/PyMuPDF/issues/1237. 

I have some PDF documents with annotations for which I need to resize the page down to A4. I used your method of convert_to_pdf but this unfortunately breaks for PDF's which use non supported fonts.

**Reply by JorjMcKie** [reply]
If you do not need annotations **_as annotation_** (similarly for form fields) but the correct visual appearance would suffice, you can bake-in these objects and convert them to permanent page content. This happens on the PDF level: `doc.bake(annots=True, widgets=True)`. The result can then be freely used in `show_pdf_page()` and `insert_pdf()` scenarios.
